src/lib/components/CustomSpeedDial.react.js

```diff
@@ -0,0 +1,98 @@
+// src/lib/components/SpeedDial.react.js
+import React from 'react';
+import PropTypes from 'prop-types';
+import Box from '@mui/material/Box';
+import SpeedDial from '@mui/material/SpeedDial';
+import SpeedDialIcon from '@mui/material/SpeedDialIcon';
+import SpeedDialAction from '@mui/material/SpeedDialAction';
+import SaveIcon from '@mui/icons-material/Save';
+import PrintIcon from '@mui/icons-material/Print';
+import ShareIcon from '@mui/icons-material/Share';
+
+/**
+ * Icon mapping for SpeedDial actions.
+ */
+const iconMap = {
```

Comment:
The user can only use those 3 icons. Is there a way to have more ?
Like all for the one from `@mui/icons-material/*`